packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -989,18 +1008,25 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     tsParseTypePredicateAsserts(): boolean {
-      if (!this.tsIsIdentifier()) {
-        return false;
-      }
-
-      const id = this.parseIdentifier();
       if (
```

Comment:
This is refactor of `tsParseTypePredicateAsserts` which aims to avoid creating node by `parseIdentifier`.